lib/Arch/Runtime/HyperCall.cpp

```diff
@@ -44,13 +44,13 @@ Memory *__remill_sync_hyper_call(State &state, Memory *mem,
     case SyncHyperCall::kX86LoadGlobalDescriptorTable:
       asm volatile("lgdt"
                    :
-                   : "=m"(__remill_read_memory_64(memory, state.addr_to_load)));
+                   : "=m"(__remill_read_memory_64(mem, state.addr_to_load)));
```

Comment:
I think the `=` in `=m` here is misplaced. See this documentation:

<img width="1504" alt="image" src="https://user-images.githubusercontent.com/43664/183769510-839392d8-5105-42f4-987a-461758a050e9.png">

From here: https://gcc.gnu.org/onlinedocs/gcc/Extended-Asm.html

It's likely that for the `lgdt`, you would want to declare a local variable, storing its value with the `__remill_read_memory_64`, then reference that variable from the inline asm.

Reply:
[fcd24b2](https://github.com/lifting-bits/remill/pull/611/commits/fcd24b236ca908a41ebe253a849ba6fc738e48c9)